### PR TITLE
feat: OpenAI embeddings with GPU based KNN

### PR DESCRIPTION
**Pull request by vonodiripsa**
Added new OpenAI embeddings Quickstart demo with GPU based KNN using NVIDIA Rapids

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

_It is a new docs notebook demonstrating usage of NVIDIA Rabids KNN on GPU._

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [x] Yes. Make sure the dependencies are resolved correctly. It depends on GPU based compute with Init Script installing NVIDIA Rapids KNN.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [x] Yes.

### CODE REVIEW

**Comment by mhamilton723** [comment]
Please clear the output in this notebook before checking it in so that the diff is minimal

**Comment by mhamilton723** [comment]
Well also want to try to get this init script to run on the databricks clusters we spin up so that it tests properly. Can you add the init script to a file in say the tools/init_scripts  directory. That way we can just link people to it, and we can upload it during the build. Well also want to add this to the GPU tests on databricks, see the nbtest folder for pointers to the GPU databricks test runner

**Comment by vonodiripsa** [comment]
Corrected the semantic prefix and added init script

**Comment by mhamilton723** [comment]
can you remove output from the notebook please?

**Comment by bvonodiripsa** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by vonodiripsa** [comment]
@microsoft-github-policy-service agree company="NVIDIA"

File: core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksUtilities.scala [hunk 1]
```diff
@@ -101,10 +101,12 @@ object DatabricksUtilities {
...
 
   val CPUNotebooks: Seq[File] = ParallelizableNotebooks
-    .filterNot(_.getAbsolutePath.contains("Fine-tune"))
+    .filterNot(_.getAbsolutePath.contains("GPU"))
     .filterNot(_.getAbsolutePath.contains("Explanation Dashboard")) // TODO Remove this exclusion
 
-  val GPUNotebooks: Seq[File] = ParallelizableNotebooks.filter(_.getAbsolutePath.contains("Fine-tune"))
+  val GPUNotebooks: Seq[File] = ParallelizableNotebooks
+    .filter(file =>
+     file.getAbsolutePath.contains("GPU"))
```
Comment (mhamilton723) [inline core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksUtilities.scala:109]: please keep fine-tine in the filternots, you had it right last time

File: core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksUtilities.scala [hunk 2]
```diff
@@ -101,10 +101,12 @@ object DatabricksUtilities {
   val ParallelizableNotebooks: Seq[File] = NotebookFiles.filterNot(_.isDirectory)
 
   val CPUNotebooks: Seq[File] = ParallelizableNotebooks
-    .filterNot(_.getAbsolutePath.contains("Fine-tune"))
+    .filterNot(_.getAbsolutePath.contains("GPU"))
```
Comment (mhamilton723) [inline core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksUtilities.scala:104]:
```suggestion
    .filterNot(_.getAbsolutePath.contains("Fine-tune"))
    .filterNot(_.getAbsolutePath.contains("GPU"))
```

File: core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksUtilities.scala (hunk 1 above)
Comment (mhamilton723) [reply core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksUtilities.scala:109]: you need to set this back to the || expression you had

File: core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksRapidsTests.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.nbtest
+
+import com.microsoft.azure.synapse.ml.nbtest.DatabricksUtilities._
+
+class DatabricksRapidsTests extends DatabricksTestHelper {
+
+  val clusterId: String = createClusterInPool(GPUClusterName, AdbGpuRuntime, 2, GpuPoolId, RapidsInitScripts)
```
Comment (mhamilton723) [inline core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/DatabricksRapidsTests.scala:10]:
```suggestion
  val clusterId: String = createClusterInPool(GPUClusterName, AdbGpuRuntime, 1, GpuPoolId, RapidsInitScripts)
```

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**